src/AcceptanceTests/Infrastructure/Cleaner.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace NServiceBus.AzureServiceBus.AcceptanceTests.Infrastructure
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Logging;
+    using Microsoft.ServiceBus;
+    using Microsoft.ServiceBus.Messaging;
+    using NServiceBus.AcceptanceTests;
+    using NUnit.Framework;
+
+    public class Cleaner : NServiceBusAcceptanceTest
+    {
+        [Test, Explicit("Intended to be executed explicitly to delete all queues and topics.")]
+        [Category("Cleanup")]
+        public async Task DeleteEntities()
+        {
+            var namespaceManager = NamespaceManager.CreateFromConnectionString(TestUtility.GetDefaultConnectionString());
+
+            var queryQueues = namespaceManager.GetQueuesAsync();
+            var queryTopics = namespaceManager.GetTopicsAsync();
+            await Task.WhenAll(queryQueues, queryTopics)
+                .ConfigureAwait(false);
+
+            var deleteQueues = queryQueues.Result.Select(queueDescription => namespaceManager.DeleteQueueAsync(queueDescription.Path));
+            var deleteTopics = queryTopics.Result.Select(topicDescription => namespaceManager.DeleteTopicAsync(topicDescription.Path));
+
+            await TryWithRetries(Task.WhenAll(deleteQueues.Concat(deleteTopics)), 5)
```

Comment:
Wouldn't we want to retry and individual queue or topic deletion operation instead of the whole thing?

Reply:
I could prune everything that is completed, leaving failed and incomplete to be retried. But I wouldn't necessarily retry individually. We still want to execute in bulk.

Reply:
Yeah but you would exactly get that. Everything that is successfully completed would be, everything that is not yet will retry or fail eventually. Things are still concurrent.